instance_counter.rb

```diff
@@ -0,0 +1,21 @@
+module InstanceCounter
+
+  def self.included(base)
+    base.extend ClassMethods
+    base.send :include, InstanseMethods
+  end
+
+  module ClassMethods
+    attr_accessor :instances
+  end
+
+  module InstanseMethods
+    protected
+    def register_instance
+      self.class.instances = 0 if self.class.instances == nil
```

Comment:
```suggestion
      self.class.instances ||= 0
```
в руби это так обычно реализуют
читается "равен себе или нолю (если в переменной пусто или фолс)"